src/server.py

```diff
@@ -69,14 +69,80 @@ def response_deconstructor(response):
     headers_split = [header.split(':', 1) for header in headers]
     headers_dict = {k.lower(): v.strip() for k, v in headers_split}
     return [protocol, status, msg, headers_dict,
-            str(len(body)), str(len(first_pass))]
+            str(len(body))]
+
+
+def request_deconstructor(request):
+    """Deconstruct request into basic components.
+    Return method, protocol and validates host header"""
+    request_msg = request
+    urequest = request_msg.decode('utf8')
+    first_pass = urequest.split(CRLF+CRLF, 1)
+    head, body = first_pass
```

Comment:
if no double-crlf, then this will raise a ValueError for not enough values to unpack.  Let's keep that in mind in case of bad requests.